.github/workflows/publish-npm-packages.yml

```diff
@@ -53,6 +53,7 @@ jobs:
               with:
                   path: publish
                   ref: wp/${{ github.event.inputs.wp_version }}
+                  fetch-depth: 100
```

Comment:
Let's comment here on why we need to specify `fetch-depth` with arbitrary number `100`.

Reply:
I increased the number to 999 and used similar comments in the workflow and the Node script. We could change the strategy if someone has a better idea, but we definitely can't rely on the period of time as a factor between releases because it's very random for minor and security WordPress releases.

Reply:
The 6.2 branch [compare/wp/6.2...trunk](https://github.com/WordPress/gutenberg/compare/wp/6.2...trunk) has 3,524, [compare/wp/6.3...trunk](https://github.com/WordPress/gutenberg/compare/wp/6.3...trunk) has 1,682 commits already

Can this option be removed or set to `0` to force a full checkout of the repo?

What would be the impact be of setting this to `0` be?

I can only think of time to check out the full full repo would be make publishing slower.


> https://github.com/actions/checkout/blob/8ade135a41bc03ea155e62e844d188df1ea18608/README.md?plain=1#L91-L93

```
    # Number of commits to fetch. 0 indicates all history for all branches and tags.
    # Default: 1
    fetch-depth: ''
```

Or could add look into getting the branch directly by adding support for `--feth-tags` and get the tag e.g `wp/6.2`
> https://github.com/actions/checkout/blob/8ade135a41bc03ea155e62e844d188df1ea18608/README.md?plain=1#L95-L97

```
    # Whether to fetch tags, even if fetch-depth > 0.
    # Default: false
    fetch-tags: ''
```

Reply:
We create `wp/X.Y` branches from the last Gutenberg plugin release before Beta 1, so it's coming from `release/X.Y`. The very first publishing from every `wp/X.Y` branch by design should ignore the history as we need to publish all packages with the dist-tag `wp-X.Y`. In effect, we only care about subsequent bug fix releases targeting WP core which is either one commit that merges cherry-pick from trunk, or every bug fix cherry-picked individually. Definitely the number isn't so big.

> What would be the impact be of setting this to 0 be?

That's always an option, but it means you need to download the entire repository consisting of:

<img width="253" alt="Screenshot 2023-10-13 at 12 36 58" src="https://github.com/WordPress/gutenberg/assets/699132/01cd0285-7265-4a14-8445-bdc2d8971bdd">